Dockerfile

```
@@ -1,42 +1,35 @@
+ENTRYPOINT ["/usr/bin/tini", "--"]
+
+# Copy built JARs from the builder stage
+COPY --from=builder /app /app
+
 EXPOSE 5000
-# Start MySQL, wait for it with a 5-minute timeout, then run JARs
-CMD ["sh", "-c", "service mysql start && timeout 300s bash -c 'until mysqladmin ping -h localhost --silent; do echo \"Waiting for MySQL...\"; sleep 2; done' && java -jar eureka-server/target/*.jar & java -jar online-classroom-management/target/*.jar & java -jar classroom-microservice/target/*.jar & java -jar course-microservice/target/*.jar & java -jar task-microservice/target/*.jar & java -jar task-submission-service/target/*.jar & java -jar gateway/target/*.jar"]
\ No newline at end of file
+
+# Run all services in background & wait
+CMD java -jar eureka-server/target/*.jar & java -jar online-classroom-management/target/*.jar & java -jar classroom-microservice/target/*.jar & java -jar course-microservice/target/*.jar & java -jar task-microservice/target/*.jar & java -jar task-submission-service/target/*.jar & java -jar gateway/target/*.jar && wait
```

Hadolint (2.12.0)

[warning] Use arguments JSON notation for CMD and ENTRYPOINT arguments

(DL3025)